### ISSUE TITLE
Revisit the relationship between members, rows and other entries.

### ISSUE THREAD

**Issue by mattmathis** [mattmathis/scvfa-webtools#2]
Consider DB rows for:
- All members
- Non-members who need online access e.g. Matt, Dinah
- Grattis Fiddler's Rags
- Secondaries and children who want to publish email, addresses or phone numbers

These probably require the introduction  new statuses, to indicate non-paying entities

Assume visibility flags only apply to the primary member.   
Secondaries & children who want to be listed in a future directory should have their own rows in the DB.

**Comment by mbryant412** [comment]
All members will be included in directory when that info is collected from members
These don't need to be in directory but are visible in WP dashboard users list
These are already being handled through the Mail Chimp audience. If you want them added to PMP enroll as a new member on the website with basic PMP info.
Secondary member are First/Last Names, Email & Phone - Children are First/Last Names only.

**Comment by mattmathis** [comment]
The specific problems with the current structure (one member row per household) are that:

- There are zero provisions for updating secondary members in Mail Chimp.   In the current configuration PMP is not configured to automatically remove addresses from MC, so previously added secondaries remain.
- It is extremely unlikely that logic can be added to subscribe multiple email addresses per row because this would require replicating the MC management logic for each potential secondary.

The alternative (one member row for each primary and secondary member) requires adding logic to coordinate membership status across multiple rows when households renew or lapse.

**Comment by mbryant412** [comment]
Are you saying you want secondary members added to Mail Chimp?

**Comment by mattmathis** [comment]
Yes, all members, primary and secondary, should be able to op-in to receive the newsletter.

**Comment by mbryant412** [comment]
Can’t be done without programming through PMP since Primary is the only payee for membership. Two solutions: 

Possible hook that lets you define another level such as secondary for MailChimp however, this would be automatic and not optional.

Create an optin MailChimp form for secondary members in the Primary dashboard.

Add optin checkbox for secondary to Directory questions.

Note, the above customizations would be charged to SCEA per our maintenance agreement.

I’m assuming that all current secondaries are in mail chimp and getting newsletters and you’re only concerned about new secondary members? 

Is it not possible to ask the Primary to share their pdf with their secondary?

Thank you,

Michael

(Edited to trim extraneous email junk - MM)

**Comment by mattmathis** [comment]
It seems to me that this is exactly what "group membership" should do: support a design where we have one row per person, that are grouped by the primary member.   e.g. row for people, groups for households.

**Comment by mbryant412** [comment]
[members_list formatted.xlsx](https://github.com/mattmathis/scvfa-webtools/files/14628518/members_list.formatted.xlsx)
I sent this to Ray.

**Comment by mattmathis** [comment]
This is cool.  Almost exactly the way I would have done it.  I'm going to move it to a new issue though for further discussion.   See https://github.com/mattmathis/scvfa-webtools/issues/18#issue-2190901931
